proposals/low-level-struct-improvements.md

```diff
@@ -410,6 +410,77 @@ Detailed Notes:
     - The core library contains the feature flag indicating support for `ref` fields
     - The `langversion` value is 11 or higher
 
+### Syntax
+[12.6.2 Local variable declarations](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/statements.md#1262-local-variable-declarations): added `'scoped'?`.
+```antlr
+local_variable_declaration
+    : 'scoped'? local_variable_mode_modifier? local_variable_type local_variable_declarators
+    ;
+
+local_variable_mode_modifier
+    : 'ref' 'readonly'?
+    ;
+```
+
+[12.9.4 The `for` statement](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/statements.md#1294-the-for-statement): added `'scoped'?` _indirectly_ from `local_variable_declaration`.
+
+[12.9.5 The `foreach` statement](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/statements.md#1295-the-foreach-statement): added `'scoped'?`.
+```antlr
+foreach_statement
+    : 'foreach' '(' 'scoped'? local_variable_type identifier 'in' expression ')'
+      embedded_statement
+    ;
+```
+
+[11.6.2 Argument lists](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#1162-argument-lists): added `'scoped'?` for `out` declaration variable.
+```antlr
+argument_value
+    : expression
+    | 'in' variable_reference
+    | 'ref' variable_reference
+    | 'scoped'? 'out' local_variable_type? identifier
+    ;
+```
+
+[--.-.- Deconstruction expressions](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md):
+```antlr
+[TBD]
```

Comment:
Did you want to address this before merge?

Reply:
I'd rather handle this in a separate PR. Deconstruction expressions seem distinct from the other cases here, and the this syntax may depend on whether we have a common form for 'out' variables and deconstruction expressions in general in the spec.